app/src/test/java/com/github/se/project/ui/map/getCameraPositionForBothLocationsTest.kt

```diff
@@ -0,0 +1,59 @@
+import com.github.se.project.ui.components.adjustZoomBasedOnDistance
+import com.github.se.project.ui.components.calculateDistance
+import com.github.se.project.ui.components.getCameraPositionForBothLocations
+import com.google.android.gms.maps.model.LatLng
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class CameraUtilsTest {
+
+  // Helper locations for testing
+  private val lessonLocation = LatLng(40.748817, -73.985428)
+  private val nearbyUserLocation = LatLng(40.748817, -73.985500) // Nearby location
+  private val distantUserLocation = LatLng(34.052235, -118.243683) // Distant location, LA
+
+  @Test
+  fun `when userLocation is null, should return CameraPosition focused on lessonLocation with default zoom`() {
+    val cameraPosition = getCameraPositionForBothLocations(null, lessonLocation)
+    assertEquals(lessonLocation, cameraPosition.target)
+    assertEquals(10f, cameraPosition.zoom, 0.1f)
+  }
+
+  @Test
+  fun `when userLocation is nearby lessonLocation, should return high zoom level`() {
+    val cameraPosition = getCameraPositionForBothLocations(nearbyUserLocation, lessonLocation)
+    val expectedZoomLevel =
+        adjustZoomBasedOnDistance(calculateDistance(nearbyUserLocation, lessonLocation))
+    assertEquals(expectedZoomLevel, cameraPosition.zoom, 0.1f)
+  }
+
+  @Test
+  fun `when userLocation is distant from lessonLocation, should return lower zoom level`() {
+    val cameraPosition = getCameraPositionForBothLocations(distantUserLocation, lessonLocation)
+    val expectedZoomLevel =
+        adjustZoomBasedOnDistance(calculateDistance(distantUserLocation, lessonLocation))
+    assertEquals(expectedZoomLevel, cameraPosition.zoom, 0.1f)
+  }
```

Comment:
Good job for testing both cases where points are near and when they are far from each others.